controllers/argocd/argocd_controller_test.go

```diff
@@ -102,6 +102,65 @@ func TestReconcileArgoCD_Reconcile(t *testing.T) {
 	}
 }
 
+func TestReconcileArgoCD_LabelSelector(t *testing.T) {
```

Comment:
Can you add more unit tests for scenarios below?
* No label selector provided and all argocd instances are reconciled
* Multiple ArgoCD instances present with matching label present on some and absent on some. Only instances matching the label are reconciled.

Reply:
Both the scenarios are covered in the unit tests but they are directed mainly towards single instance because there is a duplicity issue while using `makeTestArgoCD()`

Reply:
We can create multiple instances and update the name/labels of the instance. Doesn't that work? Also, you could try something like:

```
a := makeTestArgoCD(func(a *argoprojv1alpha1.ArgoCD) {
		a.Name = "test1"
                a.labels = ""
	})
```

Reply:
Made the changes